### ISSUE TITLE
Purchase Product: pressing enter on the text input doesn't work on /purchase-product/jetpack_search/monthly

### ISSUE THREAD

**Issue by bluefuton** [Automattic/wp-calypso#54344]
Originally noted by @jsnmoon.

**Steps to reproduce the issue**

1. Attempt to purchase a Search plan via `/purchase-product/jetpack_search/monthly`.
2. Choose a site, then try and press Enter to submit the form.

**What I expected**

Enter submits the form.

**What happened instead**

You can use Enter to select a site, but you can't use it to submit the form.

<img width="581" alt="Screen Shot 2021-07-07 at 15 53 31" src="https://user-images.githubusercontent.com/17325/124697771-7f7e9a00-df3b-11eb-9b34-afd648e6c1f5.png">

**Comment by supernovia** [comment]
📌 REPRODUCTION RESULTS
- Tested – Replicated

📌 ACTIONS
- Triaged
- Set priority to Normal
- It's already assigned, so leaving that as-is
